packages/client/src/views/RecordAudit/RecordAudit.tsx

```diff
@@ -211,9 +222,74 @@ export const STATUSTOCOLOR: { [key: string]: string } = {
 
 const ARCHIVABLE_STATUSES = [DECLARED, VALIDATED, REJECTED]
 
+function ReinstateButton({
+  toggleDisplayDialog,
+  refetch
+}: {
+  toggleDisplayDialog: () => void
+  refetch: ((variables?: OperationVariables) => void) | undefined
+}) {
+  const { declarationId } = useParams<{ declarationId: string }>()
+  const intl = useIntl()
+  const dispatch = useDispatch()
+  const declaration = useSelector((store: IStoreState) =>
+    selectDeclaration(store, declarationId)
+  )
+
+  if (!declaration) {
+    return <Redirect to={HOME} />
+  }
+
+  /* TODO: handle error */
+  return (
+    <Mutation<
+      MarkEventAsReinstatedMutation,
+      MarkEventAsReinstatedMutationVariables
+    >
+      mutation={
+        declaration.event === Event.Birth
+          ? REINSTATE_BIRTH_DECLARATION
+          : REINSTATE_DEATH_DECLARATION
```

Comment:
Can you add these in record audit utils

```
import { Event } from '@client/utils/gateway'
import {
  REINSTATE_BIRTH_DECLARATION,
  REINSTATE_DEATH_DECLARATION
} from '@client/views/RecordAudit/mutations'

export const MutationMapper = {
  [Event.Birth]: REINSTATE_BIRTH_DECLARATION,
  [Event.Death]: REINSTATE_DEATH_DECLARATION
}
```

Add use 
`mutation={MutationMapper[declaration.event]}`

So that in future we can just change the mapper.

Reply:
I think we can do that when there are more events in the future. For now it's not much of an issue.